File: 10708.py
N = int(input())
M = int(input())
targets = list(map(int, input().split()))

friends = [0] * (N + 1)
for i in range(M):
    cnt = 0
    B = list(map(int, input().split()))
    B.insert(0, 0)
    for j in range(1, N + 1):
        if B[j] == targets[i]:
            friends[j] += 1
            cnt += 1

    friends[targets[i]] += (N - cnt)

for friend in friends[1:]:
    print(friend)

File: 10708.cpp
#include <iostream>
using namespace std;

int* targets;
int* friends;

int main() {
	int N; cin >> N;

	int M; cin >> M;
	targets = new int[M];
	for (int i = 0; i < M; i++)
		cin >> targets[i];

	friends = new int[N + 1];
	for (int i = 1; i <= N; i++)
		friends[i] = 0;

	for (int i = 0; i < M; i++) {
		int cnt = 0;
		for (int j = 1; j <= N; j++) {
			int B; cin >> B;

			if (B == targets[i]) {
				friends[j]++;
				cnt++;
			}
		}

		friends[targets[i]] += (N - cnt);
	}

	for (int i = 1; i <= N; i++)
		cout << friends[i] << endl;

	delete[] targets;
	delete[] friends;
}
